Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -134,64 +134,6 @@ class PurchasesOrchestrator {
         }
     }
 
-    @available(iOS 12.2, macOS 10.14.4, watchOS 6.2, macCatalyst 13.0, tvOS 12.2, *)
```

Comment:
i THINK we probably still want this function, for our own internal use?

Reply:
Nothing uses it though?

Reply:
yeah -- we used to require customers to get the paymentDiscount from the productDiscount, and then purchase with the payment discount. but moving forward we were going to do that 1st step for them, so they could just call purchase(productDiscount), and we handle both getting the paymentDiscount AND purchasing, if that makes sense.

but maybe there's another place that logic lives that i'm not aware of. and either way, this will probably need some refactoring for the completion handler and all...so maybe leaving it in doesn't make sense 🤷

Reply:
we'll end up having a different method, because this one went from `SKProductDiscount` -> `SKPaymentDiscount`, and the new internal one will go from `StoreProductDiscount` -> `PromotionalOffer`. 

To recap: 

#### Original flow: 
1. dev calls getProducts
1. dev gets product discounts from product
1. dev calls `paymentDiscount(forProductDiscount:)` to get payment discount
1. dev calls purchaseProduct with the payment discount
2. sdk applies payment discount to the payment. 

#### New flow: 
1. dev calls getProducts
1. dev gets product discounts from product
1. dev calls `purchaseProduct` with product discount
1. sdk internally transforms product discount into payment discount, and applies it to the payment

Reply:
given that its logic is not trivial, I'd keep it so that we can convert it into using the new types and reuse its tests

Reply:
meaning that I'd remove only the public methods in Purchases.swift, but stuff from PurchasesOrchestrator down could probably stay to make it easier to convert it into the new flow instead of re-building it from scratch

Reply:
Added back!